### DISCUSSION TITLE
How to blocking audio playback?

### DISCUSSION THREAD

**Discussion by hansipete** [schreibfaul1/ESP32-audioI2S#287] [Q&A]
Hello,

first of all - great library. Really enjoy it! Thank you @schreibfaul1 for the effort you put in

I my project I need feedback sounds and text2speech that plays just before something else happens, e.g. a stream starts. My approach to this is something like this.

```
bool speechFinished = false;

void blockingTextToSpeech(char *txt)
{
    audio.connecttospeech(txt, "de");
    speechFinished = false;
    while (!speechFinished)
    {
        audio.loop();
    }
}

void audio_eof_speech(const char *info)
{
    speechFinished = true;
}
```

Somewhere in my program I can use `blockingTextToSpeech("Now playing station xy")` and it will wait for the audio to be finished before proceeding. Is this the "right" way to do it?

Tschüss,
Hans

**Comment by schreibfaul1** [comment]
Hello Hans,
GoogleTTS is not a reliable service. If it is called up too often, Google suspects a bot and blocks the IP address. In addition, the text is limited in length. I wouldn't run the audio.loop() in a while loop, especially since no time is wasted when there's nothing to do. Using the speechFinished flag is okay.

**Reply by RWBRWB** [reply]
I'm also playing with Googles TTS Audio features. 

Hans did your SpeechFinished flag work the way you wanted it? 

Schreibfaul1 - What is the limit on the text we can send? Ideally alittle more text would be ideal to get a short weather report created on one connection. 

Currently I can send about 200 words before it errors out. Is that a Google issue or the text variable size setting limitation?

**Reply by hansipete** [reply]
I do remember that it worked allright. Go for it! :-)

**Reply by schreibfaul1** [reply]
GoogleTTS actually has a limitation, only short sentences will work. MaryTTS could be the solution. Please try audio.connecttomarytts('Hello World', 'en_GB', 'dfki-spike'); I don't know where the limit is, but longer texts are possible.

**Reply by RWBRWB** [reply]
> GoogleTTS actually has a limitation, only short sentences will work. MaryTTS could be the solution. Please try audio.connecttomarytts('Hello World', 'en_GB', 'dfki-spike'); I don't know where the limit is, but longer texts are possible.

I will give MaryTTS a try then. Maybe that's the ticket even though it's voice does not sound near as normal as Googles TTS for shorter sentences.

**Reply by RWBRWB** [reply]
> I do remember that it worked allright. Go for it! :-)

I'll give it a go in the near future and report back.
